app/src/main/java/org/merakilearn/FileChooserActivity.kt

```diff
@@ -0,0 +1,172 @@
+package org.merakilearn
+
+import android.content.ContentResolver
+import android.content.Context
+import android.content.Intent
+import android.database.Cursor
+import android.net.Uri
+import android.os.Bundle
+import android.provider.OpenableColumns
+import android.util.Base64
+import android.view.View
+import android.webkit.MimeTypeMap
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import org.koin.android.ext.android.inject
+import org.merakilearn.core.navigator.MerakiNavigator
+import org.merakilearn.ui.ScratchActivity
+import java.io.BufferedOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStreamWriter
+
+
+class FileChooserActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_file_chooser)
+
+        val progressBar: ProgressBar = findViewById(R.id.progressBar3)
+        progressBar.visibility = View.VISIBLE
+
+        val intent = intent
+        val action = intent.action
+        val uri: Uri? = intent.data
+        val file: File?
+        val navigator: MerakiNavigator by inject()
+
+        if (action!!.compareTo(Intent.ACTION_VIEW) == 0) {
```

Comment:
avoid using '!!' operator unless you are 100% sure that it will never be null.
For ex - if someone launches this activity from some other route, then it might crash here.